libvast/src/detail/process.cpp

```diff
@@ -178,4 +178,26 @@ caf::settings get_status() {
 #endif
 }
 
+caf::expected<std::string> execute_blocking(const std::string& command) {
+  using namespace std::string_literals;
+  std::string result;
+  std::array<char, 256> buffer;
+  auto* out = ::popen(command.c_str(), "re");
+  if (!out)
+    return caf::make_error(ec::system_error,
+                           "popen() failed: "s + ::strerror(errno));
+  size_t nread = 0;
+  do {
+    nread = ::fread(buffer.data(), 1, buffer.size(), out);
+    result += std::string_view{buffer.data(), nread};
```

Comment:
This can throw with `std::bad_alloc`, which would lead to `pclose` not being called. Now obviously we do not handle allocation failures anywhere in the code base. Not sure what the proper course of action is here. Use some form of RAII wrapper, e.g., `std::unique_ptr` with a custom deleter?

Reply:
Not sure we really need to handle this, if `bad_alloc` terminates the process the open file descriptor will also be closed. And the process is expected to terminate by itself, it's not like we still need to send a signal or similar to shut it down.

Reply:
Reasoning sounds alright to me.